### DISCUSSION TITLE
Refresh form without losing data

### DISCUSSION THREAD

**Discussion by fecarrilho** [manojadams/mui-forms#108] [Q&A] [answered]
I would like to know if its possible to refresh the form component and keep the data that was previously filled on form. I'm working with some values that came from an useState hook on a custom component, but those values only update on form when I force refreshing, as it was not reactive to changes from outside form values...but every time a force refreshing I lose the filled data...can anyone help me with this issue?

**Comment by manojadams** [comment]
@fecarrilho Can you expand on the requirement. 
It is not clear how do you want to refresh the form component. With a browser refresh?
Also, you mentioned custom component. Can you attach an example or an external link. That will help.

**Comment by manojadams** [comment, marked as answer]
@fecarrilho Please refer how to update field value from outside of the form using `metaApi`:

https://mui-forms.vercel.app/meta-api
